portfolio/src/main/java/com/google/sps/data/BlogHashMap.java

```diff
@@ -0,0 +1,150 @@
+    for (BlogMessage message : messages) {
+      if (InternalTags.tagIsSupported(message)) {
+        if (!map.containsKey(message.getTag())){
+          LinkedList<BlogMessage> firstValue = new LinkedList<BlogMessage>();
+          firstValue.addFirst(message); 
+          map.put(message.getTag(), firstValue);
+          continue;
+        }
+        map.get(message.getTag()).addLast(message);
+      } else {
+        System.err.println("Tag not supported. Tag: " + message.getTag());
+      }
+    }
+  }
+  
+  // FOR ALL getMessages() METHODS: 
+  //      Each method takes a 'limit', an argument for the amount of messages requested.
+  //      Returned messages will not be greater than this amount, but could be less,
+  //      For instance, when the number of messages in the map is less than that requested.
+  //        In a case where different users assign different tags to equivalent messages,
+  //      Returned values could have duplicate messages - but for different tags.
+  //
+  //      Returned values contain the most recent BlogMessages.
+
+  // Returns a requested amount of all BlogMessages in map
+  private LinkedList<BlogMessage> getMessages(int limit) {
+    LinkedList values = new LinkedList();
+
+    for (LinkedList<BlogMessage> tagMessages : map.values()){
+      if (limit <= 0) {
+        break;
+      }
+
+      Iterator iterator;
+      int tagMessagesSize = tagMessages.size();
+
+      // If a tag contains more messages than requested
+      // Iterate from (number of messages - limit) to return the most recent BlogMessages. 
+      if (tagMessagesSize > limit) {
+        iterator = tagMessages.listIterator(tagMessagesSize - limit);
+      } else {
+        iterator = tagMessages.iterator();
+      }
+
+      while(iterator.hasNext() && limit >= 0) {
+        values.addLast(iterator.next());
+        limit--;
+      }
+    }
+    return values;
+  }
+
+  // Returns a requested amount of all BlogMessages for a specified tag.
+  private LinkedList<BlogMessage> getMessages(String tag, int limit) {
+    LinkedList values = new LinkedList();
+
+    if (map.containsKey(tag)) {
+      Iterator iterator;
+      int tagMessagesSize = map.get(tag).size();
+      if (tagMessagesSize > limit) {
+        iterator = map.get(tag).listIterator(tagMessagesSize - limit);
+      } else {
+        iterator = map.get(tag).iterator();
+      }
+      while (iterator.hasNext() && limit >= 0) {
+        values.addLast(iterator.next());
+        limit--;
+      }
+    } else {
+      System.err.println("Tag not found. Tag: " + tag);
+    }
+    return values;
+  } 
+
+  // This method takes a list of tags and a number of messages to load, and
+  // Returns a requested (or default) amount of BlogMessages under specified tag/tags.
+  // If tag/tags are not specified, a requested (or default) amount of all BlogMessages will be returned.
+  public LinkedList<BlogMessage> getMessages(List<String> tags, int limit) {
+    if (limit <= 0) {
```

Comment:
I would recommend against this. AFAIK, it isn't a common convention that if limit <= 0, then choose some other number in it's place. It is slightly more common that if limit == 0 then use no limit. 

Further, DEFAULT_LIMIT doesn't mean much from the user's point of view. I think in this case it is better to imagine you are writing a library that will be used by someone other than your pod. Why would they want the default to be 10? 

Always try to make sure default values are meaningful. If you can't come up with a meaningful default value, then it is usually better to handle that situation in another way like throwing an error, or re-defining the function.